src/inspector_agent.cc

```diff
@@ -841,13 +841,14 @@ void Agent::DisableAsyncHook() {
 
 void Agent::ToggleAsyncHook(Isolate* isolate,
                             const Global<Function>& fn) {
+  if (!parent_env_->can_call_into_js()) return;
```

Comment:
Would be better if some comments are added about when this would happen, maybe something like `The inspector agent might attempt to toggle the inspector async hooks to stop tracking async call stacks when the Node.js instance is shutting down and is not supposed to execute JS anymore. To avoid crashing we'll just make it a noop and stop tracking async call stacks here` (I am assuming it's fine to just stop tracking because we probably don't care when the instance is shutting down?)

Reply:
Although to quote my comment in https://github.com/nodejs/node/issues/27261#issuecomment-587567078 this still feels like a bug - we are doing inspector client cleanups *after* the client is being disconnected from the frontend. But I guess a fix like this wouldn't hurt.

Reply:
On a side note I think an ultimate fix would be to correct the state management of either `waiting_for_frontend_`, `waiting_for_sessions_disconnect_` or `waiting_for_resume_`, or add another one to the mix (though I have no idea how)

Reply:
I’ve added a comment that explains that if we hit this condition, no async events are going to be emitted anyway, and added a link to your comment here :+1: